nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsProvenanceReportingTask.java

```diff
@@ -0,0 +1,444 @@
+                final String platform = context.getProperty(PLATFORM).evaluateAttributeExpressions().getValue();
+                final Boolean allowNullValues = context.getProperty(ALLOW_NULL_VALUES).asBoolean();
+                final String nifiUrl = context.getProperty(INSTANCE_URL).evaluateAttributeExpressions().getValue();
+                URL url;
+                try {
+                        url = new URL(nifiUrl);
+                } catch (final MalformedURLException e1) {
+                        throw new AssertionError();
+                }
+
+                final String hostname = url.getHost();
+
+                final Map<String, Object> config = Collections.emptyMap();
+                final JsonBuilderFactory factory = Json.createBuilderFactory(config);
+                final JsonObjectBuilder builder = factory.createObjectBuilder();
+                final DateFormat df = new SimpleDateFormat(TIMESTAMP_FORMAT);
+                df.setTimeZone(TimeZone.getTimeZone("Z"));
+                StringBuilder sb = new StringBuilder();
+                sb.append('[');
+                CreateConsumer(context);
+                consumer.consumeEvents(context, (mapHolder, events) -> {
+                        for (final ProvenanceEventRecord event : events) {
+                                final String componentName = mapHolder.getComponentName(event.getComponentId());
+                                final String processGroupId = mapHolder.getProcessGroupId(event.getComponentId(),
+                                                event.getComponentType());
+                                final String processGroupName = mapHolder.getComponentName(processGroupId);
+                                JsonObject jo = serialize(factory, builder, event, df, componentName, processGroupId,
+                                                processGroupName, hostname, url, rootGroupName, platform, nodeId,
+                                                allowNullValues);
+                                sb.append(jo.toString());
+                                sb.append(',');
+                        }
+
+                });
+                sb.append(']');
+                return sb.toString();
+        }
+
+        private JsonObject serialize(final JsonBuilderFactory factory, final JsonObjectBuilder builder,
+                        final ProvenanceEventRecord event, final DateFormat df, final String componentName,
+                        final String processGroupId, final String processGroupName, final String hostname,
+                        final URL nifiUrl, final String applicationName, final String platform,
+                        final String nodeIdentifier, Boolean allowNullValues) {
+                addField(builder, "eventId", UUID.randomUUID().toString(), allowNullValues);
+                addField(builder, "eventOrdinal", event.getEventId(), allowNullValues);

```

Comment:
Should getEventID() be used for eventID?

Reply:
Ok. I will use event.GetEventId as eventId and remove eventOrdinal.